src/main/java/org/sunbird/extension/user/impl/UserProviderRegistryImpl.java

```diff
@@ -0,0 +1,178 @@
+
+    initialize();
+    // hardcoded to teacher till userType enhancement is done
+    userProfileMap.put("userType", "teacher");
+    readUser(userProfileMap);
+  }
+
+  @Override
+  public void update(Map<String, Object> userProfileMap) {
+
+    try {
+      initialize();
+      // hardcoded to teacher till userType enhancement is done
+      userProfileMap.put("userType", "teacher");
+      updateUser(userProfileMap);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  @Override
+  public void delete(Map<String, Object> userProfileMap) {
+    // TODO Auto-generated method stub
+
+  }
+
+  /** Method to initialize registry client */
+  private void initialize() {
+
+    headers.put(SunbirdExtensionConstants.CONTENTTYPE, MediaType.APPLICATION_JSON_VALUE);
+    headers.put(SunbirdExtensionConstants.ACCESSTOKEN, accessToken);
+
+    ITransformer<String> jsonToJsonldTransformer = JsonToJsonLDTransformer.getInstance();
+    ITransformer<String> jsonldToJsonTransformer = JsonldToJsonTransformer.getInstance();
+    client =
+        OpensaberClient.builder()
+            .requestTransformer(jsonToJsonldTransformer)
+            .responseTransformer(jsonldToJsonTransformer)
+            .build();
+  }
+
+  /**
+   * Adds User Entity to the Registry
+   *
+   * @param extensionMap
+   * @throws JsonProcessingException
+   * @throws IOException
+   * @throws TransformationException
+   */
+  private void addUser(Map<String, Object> userProfileMap)
+      throws JsonProcessingException, IOException, TransformationException {
+
+    try {
+      Config config = ConfigFactory.load(SunbirdExtensionConstants.USER_CREATE_MAPPING_FILE);
+      String userType = userProfileMap.get(SunbirdExtensionConstants.USERTYPE).toString();
+      Map<String, Object> userMap =
+          new TransformJsonUtil().transform(config, userProfileMap, userType);
+
+      String mainProvider = config.getString(SunbirdExtensionConstants.SUNBIRD_MAIN_PROVIDER);
+      List<Map> externalIds = (List<Map>) userMap.get(SunbirdExtensionConstants.EXTERNALIDS);
+      if (null != externalIds) {
+        for (Map externalIdDetails : externalIds) {
+          if (mainProvider.equalsIgnoreCase(
+              (String) externalIdDetails.get(SunbirdExtensionConstants.PROVIDER))) {
+            userMap.put(
+                SunbirdExtensionConstants.TEACHERID,
+                externalIdDetails.get(SunbirdExtensionConstants.ID));
+          }
+        }
+      }
+
+      /* System.out.println("userMap-->"+userMap); */
+
+      responseData =
+          client.addEntity(new RequestData<>(mapper.writeValueAsString(userMap)), headers);
+
+      Map<String, Object> responseMap =
+          mapper.readValue(responseData.getResponseData(), HashMap.class);
+      Map<String, Object> resultMap =
+          (Map<String, Object>) responseMap.get(SunbirdExtensionConstants.RESULT);
+
+      /*
+       * System.out.println("responseData-->"+responseData);
```

Comment:
Use project logger instead of sys out statements (or) remove commented sys outs.